src/reducers/courseReducer.js

```diff
@@ -0,0 +1,25 @@
+import * as types from '../actions/actionTypes';
+import initialState from './initialState';
+
+export default function courseReducer(state=initialState.courses, action){
+    switch(action.type){
+        case types.LOAD_COURSES_SUCCESS:
+        return action.courses;
+        
+        case types.CREATE_COURSE_SUCCESS :
+        return [...state,
+        Object.assign({}, action.course)];
+
+        case types.UPDATE_COURSE_SUCCESS:
+        return [
+            ...state.filter(course => course.id !== action.course.id),
+            Object.assign({}, action.course)
```

Comment:
you can do this instead = ```{ ...action.course }``` mind you, your web pack should be set to be able to use spread operators. So if it's not,  don't bother. 😄